src/test/java/org/ihtsdo/buildcloud/TestConfig.java

```diff
@@ -1,32 +1,127 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.boot.test.context.TestConfiguration;
-import org.springframework.cloud.aws.autoconfigure.context.ContextInstanceDataAutoConfiguration;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.Primary;
 import org.springframework.context.annotation.PropertySource;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.hibernate5.HibernateTransactionManager;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
 
-import javax.jms.JMSException;
-import javax.jms.Session;
+import java.io.IOException;
 
-@PropertySource("classpath:application.properties")
+@PropertySource(value = "classpath:application.properties", encoding = "UTF-8")
 @PropertySource(value = "classpath:application-test.properties", encoding = "UTF-8")
 @EnableConfigurationProperties
+@EnableJpaRepositories
 @TestConfiguration
 @SpringBootApplication(exclude = {
-		ContextInstanceDataAutoConfiguration.class,
 		HibernateJpaAutoConfiguration.class}
 )
 public class TestConfig extends Config {
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	private S3ClientFactory s3ClientFactory;
+
+	private static final String ACTIVEMQ_IMAGE = "symptoma/activemq";
+	private static final int ACTIVEMQ_PORT = 61616;
+
+	@SuppressWarnings("rawtypes")
+	@Container
+	private static final GenericContainer activeMqContainer = new GenericContainer(ACTIVEMQ_IMAGE).withExposedPorts(ACTIVEMQ_PORT);
+	static {
+		System.setProperty("aws.region", "us-east-1");
+		activeMqContainer.start();
+	}
+
+	@Primary
+	@Bean(name = "offlineS3Client")
+	@DependsOn("s3ClientFactory")
+	public S3Client s3Client(@Value("${srs.build.offlineMode}") final boolean offlineMode) {
+		return s3ClientFactory.getClient(offlineMode);
+	}
+
+	@Bean
+	public S3ClientFactory s3ClientFactory(@Value("${srs.build.s3.offline.directory}") final String directory) throws IOException {
+		s3ClientFactory = new S3ClientFactory();
+		s3ClientFactory.setOnlineImplementation(new S3ClientImpl(software.amazon.awssdk.services.s3.S3Client.builder().credentialsProvider(ProfileCredentialsProvider.create()).build()));
```

Comment:
We don't need to use online implementation for testing unless we are using local stack for S3

Reply:
This Bean configuration has been removed from TestConfig.java as redundant. @CoderMChu